src/main/java/com/amazon/ion/IonReader.java

```diff
@@ -63,6 +63,44 @@
  * {@link IonValue} hierarchy.  For example, to get the text of a symbol one
  * would use {@link #stringValue()}, mirroring {@link IonSymbol#stringValue()}.
  *
+ * <h2>Obtaining and Usage</h2>
+ * Instances of {@code IonReader} may be constructed through builders, which
+ * implement {@code IonReaderBuilder}. An {@code IonReaderBuilder} with the
+ * default configuration may be constructed as follows. This builder will
+ * construct {@code IonReader} instances which can read both text and binary
+ * Ion data.
+ * {@snippet :
```

Comment:
TIL about `@snippet`: https://www.baeldung.com/java-doc-code-snippets

This is from Java 18 and we're building with 21 so that checks out. I'd be rather tempted to try a build w/ Java 23 and go straight to Markdown javadoc, as long as we're still targeting Java 8.